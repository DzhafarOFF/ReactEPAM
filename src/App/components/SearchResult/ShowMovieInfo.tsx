import React  from 'react';
import { Movie } from './types';
import './ShowMovieInfo.scss';
import '../common/Button.scss';

const ShowMovieInfo = (props: Movie) => {
    return (
        <div className = 'movie-modal-container'>
            <div className = 'movie-modal'>
                <div className = 'movie-modal-header'>
                    <h1 className = 'search-title'>Netflixroulette</h1>
                    <button className = 'button' onClick = {() => props.action(props)}>Search</button>
                </div>
                <div className = 'movie-content-container'>
                    <img className = 'movie-image' src = {props.imageURL}/>
                    <div className = 'movie-modal-info'>
                        <div className = 'movie-content-container'>
                            <h3 className = 'movie-modal-title'>{props.title}</h3>
                            <p className = 'movie-rating'>{props.rating}</p>
                        </div>
                        <p className = 'movie-genres'>{props.genres.join(' ')}</p>
                        <div className = 'movie-content-container'>
                            <p className = 'movie-release'>{props.releaseDate.substring(0, 4)}</p>
                            <p className = 'movie-runtime'>{props.runtime} min</p>
                        </div>
                        <p>{props.description}</p>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default ShowMovieInfo;