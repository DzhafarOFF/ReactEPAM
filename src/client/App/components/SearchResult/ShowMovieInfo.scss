.movie-modal-container {
    background-image: url(../Search/netflixroulette.jpg);
    background-size: cover;
    background-color: #404040;
    background-blend-mode: overlay;
}
.movie-modal-header{
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.movie-modal{
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    padding: 40px;
    max-width: 1140px;
}
.movie-content-container{
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.movie-modal-info {
    text-transform: none;
    max-width: 600px;
    font-size: 1.5rem;
    color: #fefefe;
}
.movie-image {
    width: 300px;
}
.movie-modal-title{
    font-size: 2rem;
    text-transform: uppercase;
    color: lighten($color: #E50914, $amount: 20);
}
.movie-rating {
    border: 2px solid;
    display: inline-block;
    padding: 10px 5px;
    border-radius: 50%;
}

.movie-genres{
    margin-top: 0;
}
.movie-release,
.movie-runtime{
    font-weight: bolder;
}