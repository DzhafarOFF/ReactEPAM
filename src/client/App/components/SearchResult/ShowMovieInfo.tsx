import './ShowMovieInfo.scss';
import React, { useCallback } from 'react';
import { Link } from 'react-router-dom';
import { Movie } from '../../typings/types';
import { removeCurrentMovie } from '../../actions/setCurrentMovie';
import { useDispatch } from 'react-redux';

const ShowMovieInfo: React.FC<Movie> = (props: Movie) => {
	const dispatch = useDispatch();
	const selectSearch = useCallback((): void => {
		dispatch(removeCurrentMovie());
	}, [dispatch]);

	const dateStringLength = 4;
	const date = props.releaseDate.substring(0, dateStringLength);

	return (
		<div className = 'movie-modal-container'>
			<div className = 'movie-modal'>
				<div className = 'movie-modal-header'>
					<Link to='/' className = 'search-title'>
						<h1>{'Netflixroulette'}</h1>
					</Link>
					<Link to='/'>
						<button className = 'button' onClick = {selectSearch}>{'Search'}</button>
					</Link>
				</div>
				<div className = 'movie-content-container'>
					<img className = 'movie-image' src = {props.imageURL}/>
					<div className = 'movie-modal-info'>
						<div className = 'movie-content-container'>
							<h3 className = 'movie-modal-title'>{props.title}</h3>
							<p className = 'movie-rating'>{props.rating}</p>
						</div>
						<p className = 'movie-genres'>{props.genres.join(' ')}</p>
						<div className = 'movie-content-container'>
							<p className = 'movie-release'>{date}</p>
							<p className = 'movie-runtime'>{props.runtime} {'min'}</p>
						</div>
						<p>{props.description}</p>
					</div>
				</div>
			</div>
		</div>
	);
};

export default ShowMovieInfo;
