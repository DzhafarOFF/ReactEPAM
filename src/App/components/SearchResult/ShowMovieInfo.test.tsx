import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ShowMovieInfo from './ShowMovieInfo';
const props = {
    key: 1,
    title: 'MovieName',
    genres: ['Action','Drama'],
    releaseDate: '2018',
    imageURL: 'URL',
    rating: 9,
    description: 'Movie description',
    runtime: 120
}
describe('<ShowMovieInfo/> rendering', () => {
    const showMovieInfo = shallow(<ShowMovieInfo {...props}/>)
    it('should renders correctly',() => {
        expect(showMovieInfo).toMatchSnapshot();
    })
});