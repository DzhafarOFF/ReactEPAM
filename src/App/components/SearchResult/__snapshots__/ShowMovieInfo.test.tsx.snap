// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowMovieInfo/> rendering should renders correctly 1`] = `
<div
  className="movie-modal-container"
>
  <div
    className="movie-modal"
  >
    <div
      className="movie-modal-header"
    >
      <h1
        className="search-title"
      >
        Netflixroulette
      </h1>
      <button
        className="button"
        onClick={[Function]}
      >
        Search
      </button>
    </div>
    <div
      className="movie-content-container"
    >
      <img
        className="movie-image"
        src="URL"
      />
      <div
        className="movie-modal-info"
      >
        <div
          className="movie-content-container"
        >
          <h3
            className="movie-modal-title"
          >
            MovieName
          </h3>
          <p
            className="movie-rating"
          >
            9
          </p>
        </div>
        <p
          className="movie-genres"
        >
          Action Drama
        </p>
        <div
          className="movie-content-container"
        >
          <p
            className="movie-release"
          >
            2018
          </p>
          <p
            className="movie-runtime"
          >
            120
             min
          </p>
        </div>
        <p>
          Movie description
        </p>
      </div>
    </div>
  </div>
</div>
`;
